packages/web3/src/contract/contract.ts

```diff
@@ -565,7 +567,7 @@ export class Contract extends Artifact {
   async txParamsForDeployment<P extends Fields>(
     signer: SignerProvider,
     params: DeployContractParams<P>
-  ): Promise<SignDeployContractTxParams> {
+  ): Promise<SignDeployContractChainedTxParams> {
```

Comment:
Here we are returning `SignDeployContractChainedTxParams` which contains two more fields: `publicKey` and `type`, which should be backwards compatible. 
Alternatively we can create another function to return it.

Reply:
Since we decoupled the public key from tx parameters, can we keep the return type unchanged? Looks like possible.

Public keys can be fetched in other ways if necessary.

Reply:
Even if I revert this to the original types without `type` field in the return, `npm run build` would still succeed. It seems there are some type safety issues here.